wasm/main.py

```diff
@@ -2286,94 +2263,90 @@ def spec_invoke_function_address(config, a=None):
         # 6
         instrstarend = f.code.body
         # 8
-        valn = []
-        if len(t1n) > 0:
-            valn = operand_stack[-1 * len(t1n) :]
-            del operand_stack[-1 * len(t1n) :]
+        valn = list(reversed([
+            config.value_stack.pop()
+            for _ in range(len(t1n))
+        ]))
         # 9
-        val0star = []
+        val0star: List[TValue] = []
         for valtype in tstar:
             if valtype.is_integer_type:
-                val0star += [0]
+                val0star.append(UInt32(0))
             elif valtype.is_float_type:
-                val0star += [0.0]
+                val0star.append(Float32(0.0))
             else:
                 raise Exception(f"Invariant: unkown type '{valtype}'")
         # 10 & 11
-        F += [
-            {
-                "module": f.module,
-                "locals": valn + val0star,
-                "arity": len(t2m),
-                "height": len(operand_stack),
-                "continuation": [instrstar, idx + 1, control_stack],
-            }
-        ]
-        # 12
-        blockinstrstarendend = [Block(t2m, tuple(instrstarend)), End()]
-        config["instrstar"] = blockinstrstarendend
-        config["idx"] = 0
-        config["control_stack"] = []
```

Comment:
Here's the part I was referring to about clearing the label stack.  See above how the `"continuation"` contains the old label stack (previously called control_stack).  This no longer needs to be explicitly tracked like this, however, the code doesn't currently provide strong guarantees that we can't get into a situation where there is a label on the label stack which refers to a frame that is no longer part of the frame stack.